persistence/src/main/java/com/epam/esm/dao/CommonDao.java

```diff
@@ -0,0 +1,42 @@
+package com.epam.esm.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * The interface Common dao.
+ *
+ * @param <T> the type parameter
+ */
+public interface CommonDao<T> {
+    /**
+     * Add entity to Db.
+     *
+     * @param entity the entity
+     * @return the t
```

Comment:
What is t? Please take a look into your docs and update at least _@return_ tags to be more informative